### PR TITLE
feat: vault threshold events

### PR DESCRIPTION
**Pull request by jtfirek**
Dom would like me to track threshold details for vaults in the squid and it would be much more efficient if I had an event to listen too instead of querying state.
https://github.com/interlay/interbtc-squid/pull/77#issuecomment-1464837804

### CODE REVIEW

**Comment by sander2** [comment]
You added events for the global thresholds, but vaults are also allowed to set a custom secure threshold. Don't you need an event for that as well?

**Comment by jtfirek** [comment]
I added the event for the custom secure threshold. But I am not sure about where I placed it.
The most natural place for the event seems to be this function:
https://github.com/interlay/interbtc/blob/3508b609903dec2c55ad3981bafea08a3395ae3a/crates/vault-registry/src/lib.rs#L259
but this function uses option type for threshold value. I know I could get the value using unwrap(), but I thought there might be some potential for issues if the unwrap panicked on a None type. Am I correct to assume this?

File: crates/vault-registry/src/lib.rs [hunk 1]
```diff
@@ -771,6 +781,11 @@ impl<T: Config> Pallet<T> {
                 some_new_threshold.gt(&global_threshold),
                 Error::<T>::ThresholdNotAboveGlobalThreshold
             );
+            Self::deposit_event(Event::<T>::CustomThreshold {
+                vault_id: vault_id,
+                currency_pair: currency_pair,
```
Comment (gregdhill) [inline crates/vault-registry/src/lib.rs:786]:
```suggestion
                currency_pair: vault_id.currencies,
```

Comment (gregdhill) [reply crates/vault-registry/src/lib.rs:786]: Actually as pointed out by @bvotteler this can be removed because it is already included in the `vault_id`

**Pull request closed without merging**